tests/e2e/runners/runners.go

```diff
@@ -0,0 +1,169 @@
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+
+	// Auth plugin for GCP
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	ClusterPrefix = "tetragon-ci"
+)
+
+type SetupClusterFunc func(env.Environment) env.Func
+type PortForwardFunc func(env.Environment) env.Func
+
+type Runner struct {
+	setupCluster        SetupClusterFunc
+	installCilium       env.Func
+	installTetragon     env.Func
+	tetragonPortForward PortForwardFunc
+	hasCalledSetup      bool
+	keepExportFiles     bool
+}
+
+var DefaultRunner = Runner{
+	setupCluster: func(testenv env.Environment) env.Func {
+		return e2ehelpers.MaybeCreateTempKindCluster(testenv, ClusterPrefix)
+	},
+	installCilium: ciliuminstall.Setup(),
+	installTetragon: install.Install(install.WithHelmOptions(map[string]string{
+		"tetragon.exportAllowList": "",
+	})),
+	tetragonPortForward: func(testenv env.Environment) env.Func {
+		return helpers.PortForwardTetragonPods(testenv)
+	},
+	hasCalledSetup:  false,
+	keepExportFiles: false,
+}
+
+func NewRunner() *Runner {
+	runner := DefaultRunner
+	return &runner
+}
+
+func (r *Runner) WithKeepExportFiles(keep bool) *Runner {
+	r.keepExportFiles = keep
+	return r
+}
+
+func (r *Runner) WithSetupClusterFn(setup SetupClusterFunc) *Runner {
+	r.setupCluster = setup
+	return r
+}
+
+func (r *Runner) NoPortForward() *Runner {
+	r.tetragonPortForward = func(_ env.Environment) env.Func {
+		return func(ctx context.Context, _ *envconf.Config) (context.Context, error) {
+			klog.Info("Skipping Tetragon port forward")
+			return ctx, nil
+		}
+	}
+	return r
+}
+
+func (r *Runner) WithInstallTetragonFn(install env.Func) *Runner {
+	r.installTetragon = install
+	return r
+}
+
+func (r *Runner) WithInstallTetragon(options ...install.Option) *Runner {
+	r.installTetragon = func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		return install.Install(options...)(ctx, cfg)
+	}
+	return r
+}
+
+func (r *Runner) WithInstallCiliumFn(install env.Func) *Runner {
+	r.installCilium = install
+	return r
+}
+
+func (r *Runner) WithInstallCilium(options ...ciliuminstall.Option) *Runner {
+	r.installCilium = func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
+		return ciliuminstall.Setup(options...)(ctx, cfg)
+	}
+	return r
+}
+
+func (r *Runner) NoInstallCilium() *Runner {
+	r.installCilium = func(ctx context.Context, _ *envconf.Config) (context.Context, error) {
+		klog.Info("Skipping Cilium install")
+		return ctx, nil
+	}
+	return r
+}
+
+func (r *Runner) Setup() env.Environment {
+	if r.hasCalledSetup {
+		klog.Fatalf("Cannot call Runner.Setup() twice")
+	}
+	r.hasCalledSetup = true
+
+	cfg, err := envconf.NewFromFlags()
+	if err != nil {
+		klog.Fatalf("Failed to configure test environment: %w", err)
+	}
+	klog.Info("IMPORTANT: Tetragon e2e tests require parallel tests enabled. User preferences will be ignored.")
+	cfg = cfg.WithParallelTestEnabled()
```

Comment:
Can we have the flags or the configurations dumped before tests output ? as we do it now with tetragon, otherwise in separate PR